apollo-federation/src/compat.rs

```diff
@@ -281,10 +284,133 @@ pub fn coerce_schema_default_values(schema: &mut Schema) {
     }
 }
 
+fn coerce_directive_application_values(
```

Comment:
fly by thought: this looks slightly different from input coercion we do in apollo-compiler/spec, which is making me think why JS QP ever wanted to do a different a different form of input coercion. i feel like the router and the js qp ended up doing it twice _and_ did it differently 🤔 🤔 ... anyways